### ISSUE TITLE
kubectl apply set-last-applied does not use manifest's `metadata.namespace`

### ISSUE THREAD

**Issue by migueloller** [kubernetes/kubectl#787]
### What happened
Running `kubectl apply set-last-applied` after successfully running `kubectl apply` fails with message: `Error from server (NotFound): ...`. The reason for this is that while `kubectl apply` uses the manifests' `metadata.namespace` to update the resources, `kubectl apply set-last-applied` uses the default namespace for the current context of the kubeconfig.

### Expected Behavior
`kubectl apply set-last-applied` should use the manifests' `metadata.namespace` just like `kubectl apply` does.

#### `test-svc.yaml`
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: test
---
apiVersion: v1
kind: Service
metadata:
  name: test-svc
  namespace: test
spec:
  type: ExternalName
  externalName: www.example.com
```

```bash
kubectl config set-context --current --namespace=default
kubectl apply -f ./test-svc.yaml
kubectl apply set-last-applied -f ./test-svc.yaml
```

**Comment by seans3** [comment]
Thanks for the reproduction steps--very useful. Could you add the version of the client and server (run `kubectl version`). Thanks.

/sig cli
/area kubectl
/kind bug
/priority P1
/assign

**Comment by migueloller** [comment]
#### `kubectl version`
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-13T11:52:32Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
Note: Reproduction was done in minikube.
#### `minikube version`
```
minikube version: v1.6.1
commit: 42a9df4854dcea40ec187b6b8f9a910c6038f81a
```

**Comment by zhouya0** [comment]
Actually you can use specify `kubectl apply set-last-applied -f ./test-svc.yaml -n test ` to solve this problem.

But if you want to change this command without specify `-n`. Perhaps we have to change code:
https://github.com/kubernetes/kubernetes/blob/42fe74cd2c2d09a5fa80976bb82c91caa778738e/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_set_last_applied.go#L209
`o.namespace` into `info.namespace`

**Comment by migueloller** [comment]
@zhouya0,

> Actually you can use specify kubectl apply set-last-applied -f ./test-svc.yaml -n test to solve this problem.

That is correct. This is a simple example but manifest files may contain resources that are deployed to different namespaces. In this case, specifying `--namespace` would result in an error. I think it's also fair to expect `kubectl apply last-applied-configuration` to behave like `kubectl apply`.

> But if you want to change this command without specify -n. Perhaps we have to change code:
https://github.com/kubernetes/kubernetes/blob/42fe74cd2c2d09a5fa80976bb82c91caa778738e/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_set_last_applied.go#L209
o.namespace into info.namespace

Seems easy enough. Happy to open a PR if it's something that would be welcome.

**Comment by ricochet** [comment]
This will be resolved by [server-side apply](https://github.com/kubernetes/enhancements/issues/555).

**Comment by brianpursley** [comment]
Confirmed this is resolved in kubectl 1.18.0.

Using kubectl 1.17.3:
```
$ kubectl apply set-last-applied -f ./test-svc.yaml
namespace/test configured
Error from server (NotFound): services "test-svc" not found
```

Using kubectl 1.18.0:
```
$ kubectl apply set-last-applied -f ./test-svc.yaml
namespace/test configured
service/test-svc configured
```

**Comment by brianpursley** [comment]
Closing because this has been fixed.
/close
